task_8_9_12.py

```diff
@@ -0,0 +1,51 @@
+import random
+from utils import print_intro
+
+print_intro()
+
+
+ROCK = 'rock'
+PAPER = 'paper'
+SCISSORS = 'scissors'
+CHOISES = [ROCK, PAPER, SCISSORS]
+TIE = 'tie'
+
+
+def get_user_choice():
+    while True:
+        choice = input('Choose rock, paper, or scissors: ')
+        if choice in CHOISES:
+            return choice
+        print('Invalid input. Please choose rock, paper, or scissors.')
+
+
+# pylint: disable=too-many-boolean-expressions
+def determine_winner(user_choice, computer_choice):
+    if user_choice == computer_choice:
+        return TIE
+    if (user_choice == ROCK and computer_choice == SCISSORS) or \
+            (user_choice == PAPER and computer_choice == ROCK) or \
+            (user_choice == SCISSORS and computer_choice == PAPER):
```

Comment:
you can create tuple of win combinations and just check if choices match one of the wining combinations

if (user_choice, computer_choice) in WIN_COMBINATIONS: